libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -243,6 +243,24 @@ void ContractLevelChecker::checkIllegalOverrides(ContractDefinition const& _cont
 
 	checkModifierOverrides(funcSet, modSet, _contract.functionModifiers());
 
+	for (auto const* stateVar: _contract.stateVariables())
+	{
+		if (!stateVar->isPublic())
+			continue;
+
+		for (
+			auto it = find_if(funcSet.begin(), funcSet.end(), MatchByName{stateVar->name()});
```

Comment:
What if the base has multiple functions of the same name and only one / none matching the signature?